src/libs/TransactionUtils.ts

```diff
@@ -867,51 +867,73 @@ function compareDuplicateTransactionFields(transactionID: string): {keep: Partia
         reimbursable: ['reimbursable'],
     };
 
-    const getDifferentValues = (items: Array<OnyxEntry<Transaction>>, keys: Array<keyof Transaction>) => [
-        ...new Set(
-            items
-                .map((item) => {
-                    // Prioritize modifiedMerchant over merchant
-                    if (keys.includes('modifiedMerchant' as keyof Transaction) && keys.includes('merchant' as keyof Transaction)) {
-                        // eslint-disable-next-line @typescript-eslint/prefer-nullish-coalescing
-                        return item?.modifiedMerchant || item?.merchant;
-                    }
-                    return keys.map((key) => item?.[key]);
-                })
-                .flat(),
-        ),
-    ];
+    function getDifferentValues(items: Array<OnyxEntry<Transaction>>, keys: Array<keyof Transaction>) {
```

Comment:
Can you add a comment to this function as well to easily understand what we are doing?